src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -855,6 +872,76 @@ func validateListener(
 				external.Subdomain,
 				"TLS requires specifying a subdomain"))
 	}
+	if tlsEnabled && clientCACertRef != nil {
+		if !requireClientAuth {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("requireClientAuth"),
+					requireClientAuth,
+					"Enabled has to be set to true for RequireClientAuth if ClientCACertRef is set"))
+		}
+
+		if clientCACertRef.Name == "" {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("clientCACertRef"),
+					clientCACertRef,
+					"Name must be provided if ClientCACertRef is set"))
+		}
+
+		if clientCACertRef.Kind != "" && !strings.EqualFold(clientCACertRef.Kind, "Secret") {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("clientCACertRef"),
+					clientCACertRef,
+					"Kind must be set to secret if set in ClientCACertRef"))
+		}
+
+		if len(allErrs) == 0 {
+			if clientCACertRef.Namespace == "" {
+				clientCACertRef.Namespace = clusterNamespace
+			}
```

Comment:
Same question as earlier. I'd prefer we only allow secrets in the same namespace as the Cluster resource.

Reply:
Changed to have secret in the same namespace.